lib/rest.js

```diff
@@ -13,7 +13,7 @@
 
  const httpClient = require(CONSTANTS.LIBDIR + "/httpclient.js");
  
- async function post(host, port, path, headers, req, timeout, callback) {
```

Comment:
post is HTTP not HTTPS so it can't possibly need sslObj! Applies to all HTTP methods.

Second would this not break the existing code? If it would then you need to fix all samples which would break.